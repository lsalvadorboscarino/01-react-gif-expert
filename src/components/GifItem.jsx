import PropTypes from 'prop-types'
import React from 'react'

export const GifItem = ({ title, url }) => {
    return (
        <div className='card'>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

//------ PROPIEDADES -----------//
GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
