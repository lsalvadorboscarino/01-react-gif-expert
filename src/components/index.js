//ARCHIVO DE BARRIL
export * from './AddCategory';
export * from './GifGrid';
export * from './GifItem';