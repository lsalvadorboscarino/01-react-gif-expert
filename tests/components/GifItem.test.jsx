import { getByRole, render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en GifItem', () => {
    const title = 'Esto es un título'
    const url = 'url.com'

    test('Debe corresponder con el snapshoot', () => {
        const { container } = render(<GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot()
    })

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GifItem title={title} url={url} />)

        //expect(screen.getByRole('img').src).toContain(url);
        const { src, alt } = screen.getByRole('img')
        expect(src).toContain(url)
        expect(alt).toBe(title)
    })

    test('debe de mostra el título en el componente', () => {
        render(<GifItem title={title} url={url} />)

        expect(screen.getByText(title)).toBeTruthy()
    })
})