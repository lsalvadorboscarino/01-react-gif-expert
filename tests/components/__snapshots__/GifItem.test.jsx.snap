// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Debe corresponder con el snapshoot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Esto es un título"
      src="url.com"
    />
    <p>
      Esto es un título
    </p>
  </div>
</div>
`;
